### PR TITLE
feat(react): update to support react 17

### LINKED ISSUE #1473: Update to `React 17`
Once #1159 is complete, we should upgrade as a half-step to React 17. This should include both `react` and `react-dom` for development dependencies in the following packages:

- `./` (root)
- `packages/react`

No changes needed for `peerDependencies` until React 18 (see #631)

Also see https://legacy.reactjs.org/blog/2020/08/10/react-v17-rc.html for a reference of changes that may be needed. Running the newly updated tests should be sufficient to see if any specific issues are caught as the result of the upgrade.

### PR DESCRIPTION
**Pull request by scurker**
closes #1473 

The `Modal` component works fine with the Cauldron 17 upgrade.

https://github.com/user-attachments/assets/c1f4aebc-d4e2-4f46-b7f3-72fd08b51bd6

TODO:
- The OptionsMenu does not close once opened https://github.com/dequelabs/cauldron/issues/830
- Popover component https://github.com/dequelabs/cauldron/issues/1574

### CODE REVIEW

File: packages/react/src/components/Combobox/Combobox.test.tsx [hunk 1]
```diff
@@ -515,6 +515,9 @@ test('should prevent default with enter keypress and open listbox', () => {
     // rtl doesn't let us mock preventDefault
     // see: https://github.com/testing-library/react-testing-library/issues/572
     event.preventDefault = preventDefault;
+    if (type === 'focus') {
```
Comment (scurker) [inline packages/react/src/components/Combobox/Combobox.test.tsx:518]: The changes here I suspect are related to https://github.com/testing-library/user-event/issues/592, but we're not using `user-event` but are calling `fireEvent` due to specific limitations with testing library.

**Comment by anastasialanz** [comment]
Also closes https://github.com/dequelabs/cauldron/issues/1473

**Comment by scurker** [comment]
> Also closes #1473

AHHHHHHH auto completion got me and I didn't check the ticket that this one was supposed to close 😅

**Comment by anastasialanz** [comment]
Closing this for https://github.com/dequelabs/cauldron/pull/1680 since I'm taking over the work and then @scurker can be a reviewer of the PR too.

**Pull request closed without merging**